lambda_handlers/docker_authorizer.py

```diff
@@ -0,0 +1,84 @@
+'''
+    Fetch docker token from registry for hub pull and push operation for docker hub
+    Token can be alternately obtained by curl:
+    curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:jina-ai/jina:pull,push"
+'''
+import re
+import logging
+import urllib3
+import requests
+
+
+def get_logger(context='generic', file=True):
+    logger = logging.getLogger(context)
+    logger.setLevel(logging.DEBUG)
+    return logger
+
+
+def fetch_docker_token():
+    logger = get_logger(context='fetch_docker_token')
+    http = urllib3.PoolManager()
+  
+    headers = {
+        'service': DockerRegistry.SERVICE,
+        'scope': DockerRegistry.SCOPE
+    }
+    response = requests.get('https://auth.docker.io/token?service=registry.docker.io&scope=repository:jina-ai/jina:pull,push')
+    
+    logger.info(f'Got the following response status from DockerRegistry: {response.status_code}')
+    if response.status_code == 200:
+        return response.json()['token']
+    else:
+        return None
+
+def validate_docker_token(token):
+    logger = get_logger(context='validate_docker_token')
+    http = urllib3.PoolManager()
+    if not token:
+        logging.warning(f'No tokens passed to validate_docker_token')
+        return False
+    headers = {
+        'Authorization': f'Bearer {token}'
+    }
+    response = requests.get(url='https://registry.docker.io/v2/jina-ai/jina', headers=headers)
+    logger.info(f'Got the following response status from DockerRegistry: {response.status_code}')
+    if response.status_code == 200:
+        return True
+    else:
+        return False
+
+class DockerRegistry:
+    URL = 'https://auth.docker.io/token'
+    METHOD = 'GET'
+    SERVICE = 'registry.docker.io'  # docker.pkg.github.com
+    SCOPE = 'repository:jina-ai/jina:pull,push'
+    APP = 'jina-hub-api'
+
+
+class HttpVerb:
+    GET = 'GET'
+    POST = 'POST'
+    ALL = '*'
+
+
+def lambda_handler(event, context):
+    logger = get_logger(context='docker_authorizer')
+    ''' Trigger AWS event for docker auth '''
+    # if 'authorizationToken' in event:
+    #     docker_token = event['authorizationToken']
+    #     logger.info(f'Got an authorization token in the request!')
+    # else:
+    #     docker_token = ''
+    #     logger.info(f'No authorization token in the request!')
+    
+    # if 'methodArn' in event:
+    #     method_arn = event['methodArn']
+    #     logger.info(f'Current Method ARN: {method_arn}')
+    
+    token = fetch_docker_token()
+    print(str(token))
+    if validate_docker_token(token):
+        logger.info('validated docker token')
+    else:
+        logger.info('docker token not valid')
+    
```

Comment:
Please add a `return` statement so that `API Gateway` can understand the status of the lambda function execution. In this case, it should return the token in a format like [this](https://github.com/jina-ai/cloud-ops/blob/master/lambda_handlers/hubapi_list.py#L121)

Reply:
done